ecommerce/views.py

```diff
@@ -187,9 +196,90 @@ def get_object(self):
         return basket
 
 
-class CouponView(APIView):
+class BulkEnrollCouponListView(APIView):
+    """
+    Admin view for fetching coupons that can be used for bulk enrollment
+    """
+
+    permission_classes = (IsAdminUser,)
+    authentication_classes = (SessionAuthentication,)
+
+    def get(self, request, *args, **kwargs):  # pylint: disable=missing-docstring
+        serialized = [
+            {
+                **CurrentCouponPaymentSerializer(coupon_payment).data,
+                "products": ProductSerializer(
+                    (product_coupon.product for product_coupon in product_coupons),
+                    many=True,
+                ).data,
+            }
+            for coupon_payment, product_coupons in get_full_price_coupon_product_set()
+        ]
+
+        return Response(status=status.HTTP_200_OK, data=serialized)
+
+
+class BulkEnrollmentSubmitView(APIView):
+    """
+    Admin view for submitting bulk enrollment requests
+    """
+
+    permission_classes = (IsAdminUser,)
+    authentication_classes = (SessionAuthentication,)
+
+    def post(self, request, *args, **kwargs):
+        """View to send an enrollment email to all users in an uploaded csv file"""
+        product_id = request.data["product_id"]
+        coupon_payment_id = request.data["coupon_payment_id"]
+        coupon_payment = CouponPayment.objects.get(pk=coupon_payment_id)
+
+        # Extract emails of users to enroll from the uploaded user csv file
+        users_file = request.data["users_file"]
+        reader = csv.reader(users_file.read().decode("utf-8").splitlines())
+        emails = [email_row[0] for email_row in reader]
+
+        if len(emails) > coupon_payment.latest_version.num_coupon_codes:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    "errors": [
+                        {
+                            "users_file": "The given coupon has {} codes available, but {} users were submitted".format(
+                                coupon_payment.latest_version.num_coupon_codes,
+                                len(emails),
+                            )
+                        }
+                    ]
+                },
+            )
+
+        product_coupons = (
+            CouponEligibility.objects.select_related("product")
+            .select_related("coupon__payment")
+            .filter(
+                coupon__enabled=True,
+                coupon__payment=coupon_payment_id,
+                product__id=product_id,
```

Comment:
Not sure if this would be part of #188 or not, but probably should exclude coupons that have already been sent with something like:
```python
.exclude(coupon__in=BulkEnrollmentDelivery.objects.all().values_list(product_coupon, flat=True))
```

Reply:
edit, this seems to worK:
```python
        product_coupons = (
            CouponEligibility.objects.select_related("product")
            .select_related("coupon__payment")
            .filter(
                coupon__enabled=True,
                coupon__payment=coupon_payment_id,
                product__id=product_id,
            )
            .exclude(id__in=BulkEnrollmentDelivery.objects.all().values_list("product_coupon", flat=True))
        )

        if len(emails) > len(product_coupons):
            return Response(
                status=status.HTTP_400_BAD_REQUEST,
                data={
                    "errors": [
                        {
                            "users_file": "The given coupon has {} codes available, but {} users were submitted".format(
                                len(product_coupons),
                                len(emails),
                            )
                        }
                    ]
                },
            )
```